app/database/models/server/post.ts

```diff
@@ -115,10 +119,19 @@ export default class PostModel extends Model {
     /** channel: The channel which is presenting this Post */
     @immutableRelation(CHANNEL, 'channel_id') channel!: Relation<ChannelModel>;
 
+    /** thread : Query returning the thread data for the post */
+    @immutableRelation(THREAD, 'id') thread!: Relation<ThreadModel>;
+
     async destroyPermanently() {
         await this.reactions.destroyAllPermanently();
         await this.files.destroyAllPermanently();
         await this.draft.destroyAllPermanently();
+        await this.collections.get(THREAD).query(
```

Comment:
With the relation shouldn't this be `this.thread.destroyPermanently();`?

Reply:
It's working if the field is added as `Query<ReactionModel>` and `thread` is added as a `Relation<ThreadModel>`.

Reply:
Ok I'm super confused then. the relation is supposed to be 1:1 but here it seems is 1:N

so this should really be
```
try {
    (await this.thread.fetch())?.destroyPermanently(); // this in turn will call await this.participants.destroyAllPermanently(); from the thread model
} catch {
    // there is no thread record for this post
}
```

the reason for the try catch is because the schema does not have the `thread` as optional cause their keys is the table `id` column

Reply:
https://community.mattermost.com/core/pl/zzkugzz6cfgpbrt33ud4xedd3y

About the relation, My understanding was that we need to make this 1:N just like myChannels. 

I have added the above code. It's clear now about how it is called from the post model, Thank you!